pkg/build/api/types.go

```diff
@@ -966,6 +987,9 @@ type BuildRequest struct {
 	// TriggeredBy describes which triggers started the most recent update to the
 	// buildconfig and contains information about those triggers.
 	TriggeredBy []BuildTriggerCause
+
+	// StrategyOptions contains additional strategy-specific options for the build
+	StrategyOptions *BuildStrategyOptions
```

Comment:
This seems better as per strategy options:

    DockerStrategy *DockerStrategyOptions
    SourceStrategy *SourceStrategyOptions

However, this is really starting to fray the design - build requests was never intended to grow to match the build in complexity.  The more we allow here the more we distract from Build as the intended unit of submission.